tests/unit/animations/cssTransitions.ts

```diff
@@ -0,0 +1,84 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import dispatchSyntheticEvent from '../../support/dispatchSyntheticEvent';
+import cssTransistions from '../../../src/animations/cssTransitions';
+
+registerSuite({
+	name: 'animations/cssTransistions',
+
+	'enter': {
+		'transitionend'(this: any) {
+			const dfd = this.async();
+			const div = document.createElement('div');
+			cssTransistions.enter(div, {}, 'foo');
+			assert.lengthOf(div.classList, 1);
+			assert.isTrue(div.classList.contains('foo'));
+
+			setInterval(dfd.callback(() => {
+				assert.lengthOf(div.classList, 2);
+				assert.isTrue(div.classList.contains('foo-active'));
+
+				dispatchSyntheticEvent(div, 'transitionend', { bubbles: true, cancelable: true });
+
+				assert.lengthOf(div.classList, 0);
+			}), 20);
```

Comment:
can we stub `requestAnimationFrame` for these tests? we've generally managed to avoid arbitrary timed delays in `widget-core` for the most part (in unit tests at least).